staging/src/k8s.io/component-base/logs/datapol/datapol.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// Package datapol contains functions to determine if objects contain sensitive
+// data to e.g. make decisions on whether to log them or not.
+package datapol
+
+import (
+	"reflect"
+	"strings"
+)
+
+// Verify returns a list of the datatypes embeded in the argument that can be
+// considered sensitive w.r.t. to logging
+func Verify(value interface{}) []string {
+	t := reflect.ValueOf(value)
+	if t.Kind() == reflect.Ptr {
+		t = t.Elem()
+	}
+	return datatypes(t)
+}
+
+func datatypes(v reflect.Value) []string {
+	switch v.Kind() {
+	case reflect.Slice, reflect.Array:
+		for i := 0; i < v.Len(); i++ {
```

Comment:
Why do we iterate over whole list to just remove first detected datatype? We should either assume that those types are heterogeneous or not (all elements have same type).

My understanding is that golang types are hererogenius so we can assume that checking first element should be enough.

Reply:
The issue is that the first value might have the secret cleared out (see the IsZero calls in the struct case)